components/places/sql/create_shared_schema.sql

```diff
@@ -218,3 +218,15 @@ CREATE TABLE IF NOT EXISTS moz_bookmarks_synced_tag_relation(
                            ON DELETE CASCADE,
     PRIMARY KEY(itemId, tagId)
 ) WITHOUT ROWID;
+
+-- This table is a log of actions taken during each merge. An action can be a
+-- deletion (local, remote, or both), local change (uploaded to the server),
+-- remote change with remote structure (applied to the local tree), or remote
+-- change with new structure (applied to the local tree and reuploaded). This is
+-- used for telemetry gathering and diagnostics.
+CREATE TABLE IF NOT EXISTS moz_bookmarks_synced_actions(
```

Comment:
how does this table get cleaned up? It appears it might grow without bound?

Reply:
It does grow without bound. 😞 We keep Desktop logs around for 10 days; that seems like a reasonable starting point, and easy to implement. We might also think about only writing to this table if the user has trace logging enabled, though that has the same challenge as on Desktop today, where they need to repro problems after the fact.

We could _really_ get fancy with periodically rolling up old data (and logs), and compressing them into ZIP archives that our troubleshooting app can upload later...but that's getting a little carried away.

How does evicting after 10 days during maintenance sound to you?